src/aind_metadata_mapper/open_ephys/camstim_ephys_session.py

```diff
@@ -0,0 +1,366 @@
+        return tuple(available_probes)
+
+    @staticmethod
+    def manipulator_coords(
+        probe_name: str, newscale_coords: pd.DataFrame
+    ) -> tuple[aind_data_schema.components.coordinates.Coordinates3d, str]:
+        """
+        Returns the schema coordinates object containing probe's manipulator
+        coordinates accrdong to newscale, and associated 'notes'. If the
+        newscale coords don't include this probe (shouldn't happen), return
+        coords with 0.0s and notes indicating no coordinate info available
+        """
+        try:
+            probe_row = newscale_coords.query(
+                f"electrode_group == '{probe_name}'"
+            )
+        except pd.errors.UndefinedVariableError:
+            probe_row = newscale_coords.query(
+                f"electrode_group_name == '{probe_name}'"
+            )
+        if probe_row.empty:
+            return (
+                aind_data_schema.components.coordinates.Coordinates3d(
+                    x="0.0", y="0.0", z="0.0", unit="micrometer"
+                ),
+                "Coordinate info not available",
+            )
+        else:
+            x, y, z = (
+                probe_row["x"].item(),
+                probe_row["y"].item(),
+                probe_row["z"].item(),
+            )
+        return (
+            aind_data_schema.components.coordinates.Coordinates3d(
+                x=x, y=y, z=z, unit="micrometer"
+            ),
+            "",
+        )
+
+    def ephys_modules(self) -> list:
+        """
+        Return list of schema ephys modules for each available probe.
+        """
+        newscale_coords = npc_sessions.get_newscale_coordinates(
+            self.motor_locs_path
+        )
+
+        ephys_modules = []
+        for probe_letter in self.available_probes:
+            probe_name = f"probe{probe_letter}"
+            manipulator_coordinates, notes = self.manipulator_coords(
+                probe_name, newscale_coords
+            )
+
+            probe_module = session_schema.EphysModule(
+                assembly_name=probe_name.upper(),
+                arc_angle=0.0,
+                module_angle=0.0,
+                rotation_angle=0.0,
+                primary_targeted_structure="none",
+                ephys_probes=[
+                    session_schema.EphysProbeConfig(name=probe_name.upper())
+                ],
+                manipulator_coordinates=manipulator_coordinates,
+                notes=notes,
+            )
+            ephys_modules.append(probe_module)
+        return ephys_modules
+
+    def ephys_stream(self) -> session_schema.Stream:
+        """
+        Returns schema ephys datastream, including the list of ephys modules
+        and the ephys start and end times.
+        """
+        modality = aind_data_schema_models.modalities.Modality
+
+        probe_exp = r"(?<=[pP{1}]robe)[-_\s]*(?P<letter>[A-F]{1})(?![a-zA-Z])"
+
+        def extract_probe_letter(s):
```

Comment:
make this a helper method (_extract_probe_letter) rather than nested

Reply:
done :)